tasks/kernel/build.sh

```diff
@@ -20,13 +29,17 @@ if ! common/host/fetch_cache_obj.sh "$BUILD_OBJECT_KEY" "$KBUILD_OUTPUT"; then
         cd "$SOURCE"
         # CROSS_COMPILE and ARCH are set from the dockerfile
 
+        if [[ "$CONFIG_DS_KERNEL_INSTALL_IMAGE_FILESYSTEM" == 'y' ]]; then
+            TARGETS="$TARGETS Image"
+        fi
+
         if [[ "$CONFIG_DS_KERNEL_INSTALL_ZIMAGE_FILESYSTEM" == 'y' ]]; then
-            TARGETS="$TARGETS zImage"
+            [ "${ARCH_DIR}" != "arm64" ] && TARGETS="$TARGETS zImage"
         fi
 
         if [[ "$CONFIG_DS_KERNEL_INSTALL_UIMAGE_FILESYSTEM" == 'y' ]]; then
             export LOADADDR="$CONFIG_DS_KERNEL_INSTALL_UIMAGE_LOADADDR"
-            TARGETS="$TARGETS uImage"
+            [ "${ARCH_DIR}" != "arm64" ] && TARGETS="$TARGETS uImage"
```

Comment:
I don't understand the check for != arm64 on these lines. Does arm64 not support zImage or uImage?

Reply:
arm64 only supports Image, no zImage and the like.  What I had read is that they were treating this like a fresh start on this arch on how to serve the kernel, and if its going to be compressed/decompressed, that is a function of the bootloader / EFI. It can generate a compressed kernel, Image.gz, Image.zstd, etc, but its the bootloader's responsibility to decompress it rather than being self decompressing.

Reply:
These checks should probably be at the Kconfig level so its not possible to specify an impossible option.  With these checks here I think they would silently fail, but let you turn them on.  uImage/zImage should depend on armhf/armel, and Image would depend on arm64.

Reply:
I agree with it at the distro-seed Kconfig level, only allowing an Image output. Is it even possible to generate zImage/uImage from the kernel with with aarch64 selected if they are changing that workflow?

Reply:
Makes sense. Control is via the config file and enforced by Kconfig. No need for protection here. If it's the wrong architecture and makes it here, someone would know soon enough when the target fails to generate. Will remove.

Reply:
No, they are not valid make targets.  Eg:
https://github.com/embeddedTS/prototype-linux-imx93/blob/lf-6.1.y-ts/arch/arm64/boot/Makefile#L19

It looks like the Kconfig checks are already there in distro-seed, so the only suggestion I talked to Lionel on this PR was removing those checks for arm64 on uImage/zImage since they are redundant with Kconfig.